src/S3/AmbiguousSuccessParser.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Api\Parser\AbstractParser;
+use Aws\CommandInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class AmbiguousSuccessParser extends AbstractParser
+{
+    private static $ambiguousSuccesses = [
+        'UploadPartCopy' => true,
+        'CopyObject' => true,
+        'CompleteMultipartUpload' => true,
+    ];
+
+    /** @var callable */
+    private $parser;
+
+    public function __construct(callable $parser)
+    {
+        $this->parser = $parser;
+    }
+
+    public function __invoke(
+        CommandInterface $command,
+        ResponseInterface $response
+    ) {
+        $fn = $this->parser;
```

Comment:
@mtdowling The ambiguous errors will be valid XML, they just won't be the shape expected by the RestXML parser for a copy operation. I exposed the service's error parser to this class and used that instead of SimpleXML, but the copy errors need to be handled separately from list errors, as those will return a 200 and some fraction of a valid XML document (rather than a 200 and the wrong XML document).